src/stream/src/executor/source/source_backfill_executor.rs

```diff
@@ -562,10 +585,26 @@ impl<S: StateStore> SourceBackfillExecutorInner<S> {
                                     .commit(barrier.epoch)
                                     .await?;
 
-                                yield Message::Barrier(barrier);
-
-                                if self.backfill_finished(&backfill_stage.states).await? {
-                                    break 'backfill_loop;
+                                if self.should_report_finished(&backfill_stage.states) {
+                                    // TODO: use a specialized progress for source
```

Comment:
An idea: We may report the process and total rows in the progress as well.

current: (`show jobs`)
```
  Id  |                     Statement                      | Progress
------+----------------------------------------------------+----------
 1010 | CREATE MATERIALIZED VIEW mv1 AS SELECT * FROM mv1  | 2.21%
```


expected:

```
  Id  |                     Statement                         | Processed | Total (est.) |  Progress
------+-------------------------------------------------------+-----------+----------------+---------
 1010 | CREATE MATERIALIZED VIEW mv1 AS SELECT * FROM mv1     | 221       | 1000  | 2.21% 
 1012 | CREATE MATERIALIZED VIEW mv2 AS SELECT * FROM source1 | 114514    |       | 
```

Reply:
I found https://github.com/risingwavelabs/risingwave/pull/17735 already did similar thing when `set streaming_use_snapshot_backfill = true`. It just changed the `Progress` string.

```
dev=> show jobs;
┌────┬────────────────────────────────────────────────┬───────────────────┐
│ Id │                   Statement                    │     Progress      │
├────┼────────────────────────────────────────────────┼───────────────────┤
│ 29 │ CREATE MATERIALIZED VIEW mv AS SELECT * FROM t │ Snapshot [32.08%] │
└────┴────────────────────────────────────────────────┴───────────────────┘
(1 row)
```

https://github.com/risingwavelabs/risingwave/blob/3133a72aa90a718c40c29857a6337dd89527f1cd/src/meta/src/barrier/creating_job/mod.rs#L173-L227

Reply:
Let me do this in a separate PR, because it's more complex than I expected. Since one MV may have multiple backfill upstreams, and I want to distinguish MV backfill rows and source backfill rows.